pkg/queue/cluster_queue_best_effort_fifo.go

```diff
@@ -75,9 +75,15 @@ func (cq *ClusterQueueBestEffortFIFO) Delete(w *kueue.QueuedWorkload) {
 	cq.ClusterQueueImpl.Delete(w)
 }
 
-// AddInadmissibleIfNotPresent inserts a workload that cannot be admitted into
-// the inadmissibleWorkloads in ClusterQueue, unless it is already in the queue.
-func (cq *ClusterQueueBestEffortFIFO) AddInadmissibleIfNotPresent(wInfo *workload.Info) bool {
+// RequeueIfNotPresent inserts a workload that cannot be admitted into
+// ClusterQueue, unless it is already in the queue. If inadmissible is true,
+// the workload will be put into the inadmissibleWorkloads. If not,
+// the workload will be pushed back to heap directly.
+func (cq *ClusterQueueBestEffortFIFO) RequeueIfNotPresent(wInfo *workload.Info, inadmissible bool) bool {
```

Comment:
sorry for the back-and-forth. I think using `admissible` will be more readable.

Otherwise we run into double negative situations `inadmissible=false` which can be harder to reason about, potentially leading to bugs.

Reply:
I guess it's more like "potentially admissible"

Maybe `skipped` wasn't so bad?

But `admissible` is not the same as `admitted`.

Reply:
inadmissible matches the `inadmissibleWorkloads` variable, although the double negative is not good, you can think of it as  does this workload belong to `inadmissibleWorkloads`

Reply:
We can mark workloads as inadmissible if the workloads is scheduled failed. 
But we can't mark workloads as admissible if the workloads aren't tried before.

Reply:
btw, StrictFIFO is currently super aggressive in retries, if this mode turned out to be popular, then we need to optimize it: if the workload was inadmissible, then we put it back into the queue but shouldn't pop any workloads from the clusterqueue until an event happens for example.

Reply:
> if so. We also need to distinguish the two type of workloads.

the inadmissible flag?

Reply:
o.. yes  We just need to distinguish the head of the queue if it's inadmissible or not.